devshell: Print serial console as status line, dump when interrupted

cgwalters opened the pull request:
This is a followup to the previous progress status for
https://github.com/coreos/coreos-assembler/issues/1367

Rather than trying to parse the serial console, just print it as
is, except strip control characters/newlines so that it becomes
effectively a progress bar.  This way, we don't corrupt the terminal,
and we don't fill up its history either.

And, go a step farther and dump the whole serial console in the
case where we get Ctrl-C before a successful login.  This aids
in debugging.

jlebon commented:
Tried this out, though the output is a bit messy because it seems like some control characters still get through or something and we get a mix of some newline characters and some not?

I'm pretty happy with `cosa run -c` for now, so I'm good with just giving the status quo a try for some time more to see how it feels. (The "dump on Ctrl-C" thing is really neat though; could we split that out into a separate patch?)

cgwalters commented:
> Tried this out, though the output is a bit messy because it seems like some control characters still get through or something and we get a mix of some newline characters and some not?

OK, fixed that.

> (The "dump on Ctrl-C" thing is really neat though; could we split that out into a separate patch?)

They're a bit entangled in this patch; seemed simpler to fix the bug.

jlebon commented:
/lgtm

cgwalters commented:
Ugh, I swear I fixed the "grub prompt spewed to console" but apparently not...
